rb/lib/selenium/webdriver/common/selenium_manager.rb

```diff
@@ -39,30 +39,36 @@ def bin_path
         def driver_path(options)
           command = generate_command(binary, options)
 
-          location = run(*command)
-          WebDriver.logger.debug("Driver found at #{location}", id: :selenium_manager)
-          Platform.assert_executable location
+          output = run(*command)
 
-          location
+          browser_path = output['browser_path']
+          driver_path = output['driver_path']
+          Platform.assert_executable driver_path
+
+          if options.respond_to? :binary
+            options.binary = browser_path
+            options.browser_version = nil
+          end
+
+          driver_path
         end
 
         private
 
         def generate_command(binary, options)
-          command = [binary, '--browser', options.browser_name, '--output', 'json']
+          command = [binary, '--browser', options.browser_name]
           if options.browser_version
             command << '--browser-version'
             command << options.browser_version
           end
           if options.respond_to?(:binary) && !options.binary.nil?
             command << '--browser-path'
-            command << options.binary.gsub('\\', '\\\\\\')
```

Comment:
Is this really fixed in SM now?

Reply:
I ran without it and it worked with spaces, but now that I think about it, it was there for if the user backspaced their own spaces? I'll put it back.